src/extensions/averagewaveforms/AverageWaveformsView/waveformLayer.ts

```diff
@@ -17,6 +18,24 @@ const initialLayerState = {
     electrodeBoxes: []
 }
 
+type LocalLayerProps = LayerProps & { customActions: (ActionItem | DividerItem)[] }
+
+// If any custom actions have been set (that is, something a user of this component wants to happen in response to a key press)
+// expect them to have been passed in with the key 'customActions' & call them here.
+export const handleKeyboardEvent: KeyboardEventHandler = (e: KeyboardEvent, layer: CanvasWidgetLayer<LocalLayerProps, LayerState>): boolean => {
```

Comment:
It's possible that processing custom actions could be assumed by default--so this could be part of the `CanvasWidgetLayer` rather than manually added to specific layers. (Not sure if that would result in double-processing some actions or in needless empty checks, though.)

Reply:
we'd need to be careful... i don't think it's a big problem to manually add to specific layers.